packages/block-editor/src/store/reducer.js

```diff
@@ -2560,11 +2640,16 @@ export function withDerivedBlockEditingModes( reducer ) {
 				}
 				break;
 			}
+			case 'SET_BLOCK_EDITING_MODE':
+			case 'UNSET_BLOCK_EDITING_MODE':
 			case 'SET_HAS_CONTROLLED_INNER_BLOCKS': {
-				const updatedBlock = nextState.blocks.tree.get(
+				const updatedBlock = getBlockTreeBlock(
+					nextState,
```

Comment:
The change here is because `SET_BLOCK_EDITING_MODE` is often called on the root block (with a client id of `''`). When you do `nextState.blocks.tree.get( '' )`, it returns an object that doesn't have a `clientId` property and that causes some of the later code to fail. `getBlockTreeBlock` works around that by patching the returned object to have a clientId.

It might be better to make sure the root in the tree does have a `clientId` property. 😄